### PR TITLE
fix categorical what-if in RAI dashboard

### PR DESCRIPTION
**Pull request by imatiach-msft**
## Description

Fix categorical what-if in RAI dashboard

Fixes user issue:
https://github.com/microsoft/responsible-ai-toolbox/issues/1138

There were multiple problems, including a missing function implementation and also a notebook example which created a model that was too simple and didn't show any change in the predicted value for different inputs.

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [ ] responsibleai/core-ui
- [x] responsibleai/counterfactuals
- [ ] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [ ] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [x] responsibleai/model-assessment

Python packages changed:

- [ ] raiwidgets
- [ ] responsibleai
- [ ] erroranalysis
- [ ] rai_core_flask

## Tests

- [x] No new tests required.
- [ ] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Documentation:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: libs/counterfactuals/src/lib/CounterfactualList.tsx [hunk 1]
```diff
@@ -270,19 +275,29 @@ export class CounterfactualList extends React.Component<
...
     _event: React.FormEvent<IComboBox>,
     option?: IComboBoxOption
   ): void => {
+    console.log("Updating dropdown value");
     const id = key.toString();
     const keyIndex =
       this.props.data?.feature_names_including_target.indexOf(id);
     if (option?.text) {
-      this.props.setCustomRowProperty(`Data${keyIndex}`, false, option.text);
+      console.log("Updating dropdown value - text non empty");
```
Comment (zhb000) [inline libs/counterfactuals/src/lib/CounterfactualList.tsx:289]: Remove

File: libs/counterfactuals/src/lib/CounterfactualList.tsx [hunk 2]
```diff
@@ -291,14 +306,15 @@ export class CounterfactualList extends React.Component<
     evt: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>,
     newValue?: string
   ): void => {
+    console.log("Updating col value");
```
Comment (zhb000) [inline libs/counterfactuals/src/lib/CounterfactualList.tsx:309]: this is just for debugging right?

File: libs/counterfactuals/src/lib/CounterfactualList.tsx [hunk 3]
```diff
@@ -343,7 +359,11 @@ export class CounterfactualList extends React.Component<
               allowFreeform
               selectedKey={`${this.state.data[column.key]}`}
               options={dropdownOption.data.categoricalOptions}
-              onChange={this.updateDropdownColValue.bind(this, column.key)}
+              onChange={this.updateComboBoxColValue.bind(
+                this,
```
Comment (zhb000) [inline libs/counterfactuals/src/lib/CounterfactualList.tsx:363]: can we just use arrow func instead of binding this?

**Review by zhb000 — APPROVED** [review]

File: libs/counterfactuals/src/lib/CounterfactualList.tsx (hunk 1 above)
Comment (imatiach-msft) [reply libs/counterfactuals/src/lib/CounterfactualList.tsx:289]: done!

File: libs/counterfactuals/src/lib/CounterfactualList.tsx (hunk 2 above)
Comment (imatiach-msft) [reply libs/counterfactuals/src/lib/CounterfactualList.tsx:309]: done, removed

File: libs/counterfactuals/src/lib/CounterfactualList.tsx (hunk 3 above)
Comment (imatiach-msft) [reply libs/counterfactuals/src/lib/CounterfactualList.tsx]: done

**Review by vinuthakaranth — APPROVED** [review]

**Pull request merged**